src/main/scala/io/iohk/ethereum/domain/SignedTransaction.scala

```diff
@@ -23,98 +23,94 @@ object SignedTransaction {
-      bytesToSign
-    )
   }
 
-  lazy val syntacticValidity: Boolean = {
+  def getSender(tx: Transaction, pointSign: Byte, signatureRandom: BigInt, signature: BigInt, recoveredPointSign: Int): Option[Address] = {
+    val bytesToSign: Array[Byte] = if (pointSign == negativePointSign || pointSign == positivePointSign) {
+      //global transaction
+      crypto.kec256(
+        rlpEncode(RLPList(
+          tx.nonce,
+          tx.gasPrice,
+          tx.gasLimit,
+          tx.receivingAddress.toArray,
+          tx.value,
+          tx.payload)))
+    } else {
+      //chain specific transaction
+      crypto.kec256(
+        rlpEncode(RLPList(
+          tx.nonce,
+          tx.gasPrice,
+          tx.gasLimit,
+          tx.receivingAddress.toArray,
+          tx.value,
+          tx.payload,
+          Config.Blockchain.chainId,
+          valueForEmptyR,
+          valueForEmptyS)))
+    }
+
+    val recoveredPublicKey: Option[Array[Byte]] =
+      ECDSASignature.recoverPubBytes(
+        new BigInteger(1, signatureRandom.toByteArray),
+        new BigInteger(1, signature.toByteArray),
+        ECDSASignature.recIdFromSignatureV(recoveredPointSign),
+        bytesToSign
+      )
+
+    for {
+      key <- recoveredPublicKey
+      addrBytes = crypto.kec256(key).slice(FirstByteOfAddress, LastByteOfAddress)
+      if addrBytes.length == Address.Length
+    } yield Address(addrBytes)
+  }
 
-    import tx._
-    import Transaction._
+  def apply(tx: Transaction, pointSign: Byte, signatureRandom: ByteString, signature: ByteString): Option[SignedTransaction] =
+    SignedTransaction(tx, pointSign, BigInt(1, signatureRandom.toArray), BigInt(1, signature.toArray))
 
-    def byteLength(b: BigInt): Int = b.toByteArray.length
+  def apply(tx: Transaction, pointSign: Int, signatureRandom: ByteString, signature: ByteString): Option[SignedTransaction] = {
+    if(pointSign.toByte != pointSign)
+      None
+    else
+      SignedTransaction(tx, pointSign.toByte, signatureRandom, signature)
+  }
 
-    byteLength(nonce) <= NonceLength &&
-    //FIXME: the below doesn't make sense, an exception will be thrown if the condition is not met
-    //(receivingAddress.bytes.isEmpty || receivingAddress.bytes.length == AddressLength) &&
-    byteLength(gasLimit) <= GasLength &&
-    byteLength(gasPrice) <= GasLength &&
-    byteLength(value) <= ValueLength &&
-    signatureRandom.length <= ECDSASignature.RLength &&
-    signature.length <= ECDSASignature.SLength &&
-    recoveredSenderAddress.isDefined
+  def apply(tx: Transaction, pointSign: Byte, signatureRandom: BigInt, signature: BigInt): Option[SignedTransaction] = {
+    for {
+      recoveredPointSign <- SignedTransaction.getRecoveredPointSign(pointSign)
+      sender <- SignedTransaction.getSender(tx, pointSign, signatureRandom, signature, recoveredPointSign)
+    } yield SignedTransaction(tx, pointSign, signatureRandom, signature, sender)
   }
+}
+
+case class SignedTransaction protected (
```

Comment:
`protected` will not work with case class the way you expect to, because there's a factory method with this signature which is always public - you can only truly hide constructors with regular classes. But I think it's fine to have this one public, for tests if nothing else.